presto-docs/src/main/sphinx/installation/deploy-docker.rst

```diff
@@ -48,81 +46,56 @@ verify your Docker setup by running the command in step 4.
-
-   ``docker run -p 8080:8080 --name presto ahanaio/prestodb-sandbox``
-
-   This command also assigns the name ``presto`` for ``ahanaio/prestodb-sandbox``.
-
-   The Presto server begins logging startup information in the terminal window. Once you see a response similar to the following, the Presto server is running in the Docker container.
-
-   ``======== SERVER STARTED ========``
+   Use the latest non-edge Docker image on `Presto on DockerHub <https://hub.docker.com/r/prestodb/presto/tags>`_.
 
-3. To start the Presto CLI (command-line interface), open a new terminal window and run the following command:
-
-   ``docker exec -it presto presto-cli``
-
-   The presto prompt is shown:
-
-   ``presto>``
-
-   If the ``docker exec`` command returns an error similar to the following:
-   
-   ::
-    
-    "docker exec" requires at least 2 arguments.
-    See 'docker exec --help'.
-    Usage:  docker exec [OPTIONS] CONTAINER COMMAND [ARG...]
-    Execute a command in a running container``
+   Downloading the container may take a few minutes. When the download completes, go on to the next step.
 
-   the Presto server has not completed starting. Wait a little and try again, until the ``presto>`` prompt is shown.
+2. On your local system, create a file named ``config.properties`` containing the following text: 
 
-4. To verify that your Presto server and Presto CLI are communicating together, use the Presto CLI to query your local Presto server. Run the following command at the ``presto>`` prompt:
+   .. code-block:: none
 
-   ``show catalogs;``
+    coordinator=true
+    node-scheduler.include-coordinator=true
+    http-server.http.port=8080
+    discovery-server.enabled=true
+    discovery.uri=http://localhost:8080
 
-   The response should be similar to the following:
+3. On your local system, create a file named ``jvm.config`` containing the following text: 
 
-   ::
+   .. code-block:: none
 
-    presto> show catalogs;
-    Catalog
-    ---------
-    jmx
-    memory
-    system
-    tpcds
-    tpch
-    (5 rows)
-    
-    Query 20230614_181140_00000_uutw6, FINISHED, 1 node
-    Splits: 19 total, 19 done (100.00%)
-    [Latency: client-side: 0:13, server-side: 0:11] [0 rows, 0B] [0 rows/s, 0B/s]
-    
-    presto>
+    -server
+    -Xmx2G
+    -XX:+UseG1GC
+    -XX:G1HeapRegionSize=32M
+    -XX:+UseGCOverheadLimit
+    -XX:+ExplicitGCInvokesConcurrent
+    -XX:+HeapDumpOnOutOfMemoryError
+    -XX:+ExitOnOutOfMemoryError
+    -Djdk.attach.allowAttachSelf=true
+     
+4. To start the Presto server in the Docker container, run a command similar to the following example:
 
-5. To use the Presto CLI in your local Docker image to connect to a remote Presto server, run the following command: 
+   ``docker run -p 8080:8080 -it -v </local/path/to/config.properties>:/opt/presto-server/etc/config.properties -v </local/path/to/jvm.config>:/opt/presto-server/etc/jvm.config --name presto prestodb/presto:0.284``
```

Comment:
Now that I'm thinking about it, it would be more correct to mount the volume as a directory. The instructions would have to change slightly in order to direct users to create a directory on their local system that includes the `config.properties` and `jvm.config` files above. I believe it would also have to include a minimal `node.properties` with the following content:

```
node.environment=test
```

Then the command would be simplified to:

`docker run -p 8080:8080 -it -v </path/to/local/config/dir>:/opt/presto-server/etc --name presto prestodb/presto:0.284`

This would make it possible to restart the container as needed (among other things) without erroring out

Reply:
Nice suggestion! That would be a good way to approach this, if we were going to stay with the existing state of the software. 

In the interests of time for the current issue, I'd prefer not to do this as we're looking at other approaches such as proposing changes to the Presto Docker container that would remove the need for the current doc workaround, and we may not need these additional steps at all in future. Does that make sense?

Reply:
Makes complete sense!